flexmeasures_client.py

```diff
@@ -355,25 +343,63 @@ def trigger_schedule(self, *args, **fnc_kwargs):
         if start_relaxation_window < rounded_now:
             start_relaxation_window = rounded_now
 
+        back_to_max_soc = fnc_kwargs["back_to_max_soc"]
+        self.log(f"trigger_schedule, back_to_max_soc: '{back_to_max_soc}'.")
+
+        # If back_to_max_soc is a datetime (not None) the current SoC > max-soc and the soc should be back at max-soc at this time.
+        # It is set and reset in v2g_liberty set_next_action()
+        soc_maxima = []
+        if isinstance(back_to_max_soc, datetime):
+            if back_to_max_soc >= start_relaxation_window:
+                # There is a calendar item before the back_to_max_soc, so for now the latter is ignored.
+                # The only thing that changes is the setting for the soc maxima, instead of the usual CAR_MAX_SOC_IN_KWH
+                # the current SoC is used.
+                # The relaxation window is still based on the CAR_MAX_SOC_IN_KWH.
+                soc_maxima = [
+                    {
+                        "value": current_soc_kwh,
+                        "datetime": dt.isoformat(),
+                    } for dt in [rounded_now + x * resolution for x in range(0, (start_relaxation_window - rounded_now) // resolution) ]
+                ]
+            else:
```

Comment:
Let's have explicit scenarios for these cases:
- Start of calendar item's charging relaxation window is right of back_to_soc_max: `A < B < C`
- Start of calendar item's charging relaxation window is right of now, but left of back_to_soc_max: : `A < C < B`
- Start of calendar item's charging relaxation window is left of now: `C < A < B`